code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -63,6 +63,8 @@
 	return TRUE
 
 /mob/living/simple_animal/hostile/megafauna/death(gibbed, var/list/force_grant)
+	if(SSticker.mode)
+		SSticker.mode.check_win() //Calls the rounds wincheck, to see if all megafauna have died
```

Comment:
you shouldn't call it for every single mode needlessly

Reply:
this should be a signal

Reply:
then your gamemode can register for the signal from each of the megafauna at roundstart

Reply:
pretty sure this is how it’s checked for other gamemodes but i think there’s also another proc used in ops that doesn’t involve shoving these around

Reply:
that DOES NOT MAKE IT RIGHT

Reply:
literally how every other gamemode does it so

Reply:
you can keep repeating how bad the rest of them are at doing it, that won't change how it should be done

Reply:
im telling you that he shouldn't be putting the check there in the first place. there's talk of phasing out check_win in the first place.

if he does something like locate(megafauna_type) in glob.megafauna_mobs or something in check_finished then why not